lib/utils/filters.js

```diff
@@ -53,39 +57,44 @@ const validateFilters = function(type, filterObj) {
  * @param {Object} filterObj A simple key-value pair object for setting filters.
  * @returns {String} Returns the query header if the filter is valid. Else, it will throw an error.
  */
-const filterBuilder = function(searchType, filterObj=null) {
-    if (filterObj) { validateFilters(searchType, filterObj); }
+const filterBuilder = function (searchType, filterObj = null) {
+	if (filterObj) {
+		validateFilters(searchType, filterObj);
+	}
 
-    let baseQuery = `query ($page: Int, $perPage: Int, $search: String) {
+	let baseQuery = `query ($page: Int, $perPage: Int, $search: String) {
         Page (page: $page, perPage: $perPage) { pageInfo { total currentPage lastPage hasNextPage perPage } `;
 
-    if (searchType === "activity") {
-        baseQuery = `query ($page: Int, $perPage: Int, $id: Int) {
+	if (searchType === "activity") {
+		baseQuery = `query ($page: Int, $perPage: Int, $id: Int) {
             Page (page: $page, perPage: $perPage) { pageInfo { total currentPage lastPage hasNextPage perPage } 
             activities (id: $id`;
-    }
-    if (searchType === "anime") {
-        baseQuery += "media (type: ANIME, search: $search";
-    }
-    if (searchType === "manga") {
-        baseQuery += "media (type: MANGA, search: $search";
-    }
-
-    if (filterObj) {
-        let filterKeys = Object.keys(filterObj);
-    
-        for (let x = 0; x < filterKeys.length; x++) {
-            let filterValue = filterObj[filterKeys[x]];
-
-            if (filterValue instanceof Array) {
-                filterValue = `[${filterValue.join(", ")}]`;
-            }
-
-            baseQuery += `, ${filterKeys[x]}: ${filterValue}`; 
-        }
-    }
-
-    return `${baseQuery}) {`;
+	} else if (searchType === "anime") {
+		baseQuery += "media (type: ANIME, search: $search";
+	} else if (searchType === "manga") {
+		baseQuery += "media (type: MANGA, search: $search";
+	}
+
+	if (filterObj) {
+		let filterKeys = Object.keys(filterObj);
+
+		for (let x = 0; x < filterKeys.length; x++) {
+			let filterValue = filterObj[filterKeys[x]];
+
+			// If a value is completely uppercase, we are going to assume it's an ENUM
+			if (filterValue instanceof Array) {
+				filterValue = `[${filterValue
+					.map((value) => typeof value === "string" && value !== value.toUpperCase() ? `"${value}"` : value)
+					.join(", ")}]`;
+			} else if (typeof filterValue === "string" && filterValue !== filterValue.toUpperCase()) {
+				filterValue = `"${filterValue}"`;
+			}
```

Comment:
Same as the previous comment. Two if statements should be good and faster than the if/else if block. We won't have to worry about getting both since an Array instance is not a string type and vice versa.